lib/assert.js

```diff
@@ -202,6 +246,11 @@ function objEquiv(a, b, strict) {
     b = pSlice.call(b);
     return _deepEqual(a, b, strict);
   }
+  if (a instanceof Map && b instanceof Map) {
```

Comment:
`instanceof` is dangerous, it doesn't work between different contexts. We can use `Object.prototype.toString` instead:

```js
> var map = vm.runInNewContext('new Map');
undefined
> map instanceof Map
false
> Object.prototype.toString.call(map)
'[object Map]'
```